ignite/contrib/handlers/param_scheduler.py

```diff
@@ -468,15 +468,28 @@ def __init__(self, schedulers, durations, save_history=False):
         self.schedulers = schedulers
         self.durations = durations
 
-        self.optimizer = self.schedulers[0].optimizer
-        if not (all(id(s.optimizer) == id(self.optimizer) for s in self.schedulers)):
+        param_schedulers = [s for s in self.schedulers if isinstance(s, ParamScheduler)]
+        param_group_schedulers = [s for s in self.schedulers if isinstance(s, ParamGroupScheduler)]
+
+        param_optimizers = [s.optimizer for s in param_schedulers]
+        param_group_optimizers = [s.optimizer for sc in param_group_schedulers for s in sc.schedulers]
+        optimizer = list(set(param_optimizers + param_group_optimizers))
+        if len(optimizer) != 1:
             raise ValueError("schedulers should be related to same optimizer")
 
+        param_names = [s.param_name for s in param_schedulers]
+        param_group_names = [s.param_name for sc in param_group_schedulers for s in sc.schedulers]
+        param_name = list(set(param_names + param_group_names))
+        if len(param_name) != 1:
+            raise ValueError("schedulers should be related to same param_name")
```

Comment:
@sdesrozis can you please add a test for that ?

Reply:
Yes !